src/Support/QueueMonitor/QueueMonitorManager.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace FluxErp\Support\QueueMonitor;
+
+use FluxErp\Contracts\ShouldBeMonitored;
+use FluxErp\Models\QueueMonitor;
+use FluxErp\States\QueueMonitor\Failed;
+use FluxErp\States\QueueMonitor\Queued;
+use FluxErp\States\QueueMonitor\Running;
+use FluxErp\States\QueueMonitor\Stale;
+use FluxErp\States\QueueMonitor\Succeeded;
+use Illuminate\Contracts\Queue\Job;
+use Illuminate\Queue\Events\JobExceptionOccurred;
+use Illuminate\Queue\Events\JobFailed;
+use Illuminate\Queue\Events\JobProcessed;
+use Illuminate\Queue\Events\JobProcessing;
+use Illuminate\Queue\Events\JobQueued;
+use Illuminate\Support\Carbon;
+
+class QueueMonitorManager
+{
+    public static function handle(JobQueued|JobProcessing|JobProcessed|JobFailed|JobExceptionOccurred $event): void
+    {
+        $method = lcfirst(class_basename($event));
+
+        if (method_exists(static::class, $method) && static::shouldBeMonitored($event->job)) {
+            static::$method($event);
+        }
+    }
+
+    protected static function jobQueued(JobQueued $event): void
+    {
+        app(QueueMonitor::class)->query()->create([
```

Comment:
why no actions for QueueMonitor Model?

Reply:
internal only, not intended to be used otherwise currently…